significantly reduce hardcoding in Ginkgo Tests

ianvernon opened the pull request:
This PR seeks to minimize the amount of hardcoding of commonly used strings in the Ginkgo tests. It also adds some more helper functions and wrappers around commonly used commands.

Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #2068 
Fixes: #2067 
Fixes: #2070

Commit:
test/helpers: add constants for use throughout tests

Add a variety of constants for use in tests to significantly reduce the amount
of hardcoding.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test: cleanup test_suite_test.go

* Utilize constants added in test/helpers/cons.go
* Add `gops` use for getting stack traces in case the Ginkgo tests deadlock.
* Miscellaneous cleanup / log message changes.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/ginkgo-ext: utilize constants

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: utilize constants / refactoring

* Use constants recently added in `test/helpers/cons.go` to reduce amount of
hardcoding.
* Refactor Node structure to be more aptly-named to SSHMeta.
* Add wrappers around commonly used testing commands (ping, curl, netperf).

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/k8sT: refactor / use constants

* Use constants recently added in `test/helpers/cons.go`
* Miscellaneous log message cleanup.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test: helpers.basePath --> helpers.BasePath

* Make base path constant public so that other packages can utilize it
* Ensure that errors in createInterface function are handled and not ignored

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test: make timeout helper public

Make timeout helper public and utilize it in PolicyImport calls

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/runtime: refactor / cleanup

* Use constants recently added in `test/helpers/cons.go`.
* Miscellaneous log message changes and cleanup.

Signed-off by: Ian Vernon <ian@cilium.io>

ianvernon commented on test/helpers/cilium.go:
```diff
@@ -317,22 +339,24 @@ func (c *Cilium) PolicyWait(revisionNum int) *CmdRes {
+	commands = []string{"cilium policy get", "cilium config"}
 	for _, cmd := range commands {
-		fmt.Fprintf(wr, "Command '%s': \n", cmd)
+		fmt.Fprintf(wr, "\nOuput of command '%s': \n", cmd)
```
Will fix typo in next upload.

ianvernon commented on test/helpers/cilium.go:
```diff
@@ -317,22 +339,24 @@ func (c *Cilium) PolicyWait(revisionNum int) *CmdRes {
 
 	//FIXME: Ginkgo PR383 add here --since option
-	res := c.Node.Exec("sudo journalctl --no-pager -u cilium")
+	res := c.Node.Exec("sudo journalctl --no-pager -u cilium | tail -n 50")
```
Will remove this, this is leftover from debugging.

ianvernon commented on test/helpers/cilium.go:
```diff
@@ -317,22 +339,24 @@ func (c *Cilium) PolicyWait(revisionNum int) *CmdRes {
+	res = c.Exec("endpoint list")
 	fmt.Fprint(wr, res.Output())
-
+	commands = []string{"cilium policy get", "cilium config"}
```
Will remove this, this is leftover from debugging.

ianvernon commented on test/helpers/cilium.go:
```diff
@@ -317,22 +339,24 @@ func (c *Cilium) PolicyWait(revisionNum int) *CmdRes {
 		fmt.Fprint(wr, res.Output())
 	}
 	fmt.Fprint(wr, "StackTrace Ends\n")
+	Sleep(120 * time.Second)
```
Will remove this, this is leftover from debugging.

ianvernon commented on test/helpers/cilium.go:
```diff
@@ -355,25 +379,10 @@ func (c *Cilium) ServiceDel(id int) *CmdRes {
-	if !res.WasSuccessful() {
-		return fmt.Errorf("%s", res.CombineOutput())
-	}
+	// TODO: no need for custom systemd configuration, so we might be able
```
TODO: remove this.

ianvernon commented on test/helpers/docker.go:
```diff
@@ -23,15 +23,15 @@ import (
 func CreateDocker(target string, log *log.Entry) *Docker {
-	log.Infof("Docker: set target to '%s'", target)
-	node := CreateNodeFromTarget(target)
+	fmt.Printf("Docker: set target to '%s'\n", target)
```
Change this to `log.Infof`

ianvernon commented on test/helpers/kubectl.go:
```diff
@@ -299,21 +299,21 @@ func (kub *Kubectl) CiliumNodesWait() (bool, error) {
 				return false
 			}
-			kub.logCxt.Infof("K8s node '%s' ipv4 addr '%s'", k, v)
+			kub.logCxt.Infof("Kubernetes node %q IPv4 addrress: %q", k, v)
```
addrress --> address

ianvernon commented on test/helpers/ssh_command.go:
```diff
@@ -24,10 +24,12 @@ import (
 	"os"
 	"strconv"
 
+	//"bufio"
```
Remove import if it isn't being used.

ianvernon commented on test/helpers/ssh_command.go:
```diff
@@ -48,7 +50,7 @@ type SSHClient struct {
 }
 
-//SSHConfig contains metadata for running an SSH session .
+//SSHConfig contains metadata for an SSH session with a Vagrant VM .
```
remove extra space after "VM"

nebril replied:
And add a space before SSHConfig :)

ianvernon commented on test/helpers/ssh_command.go:
```diff
@@ -57,18 +59,18 @@ type SSHConfig struct {
 }
 
-//SSHConfigs map with all sshconfig. Key represent the virtualserver target name
+// SSHConfigs maps the name of a host (VM) to the SSHConfiguration
```
to its corresponding SSHConfig

ianvernon commented on test/helpers/ssh_command.go:
```diff
@@ -95,6 +105,7 @@ func (cfg *SSHConfig) GetSSHAgent() ssh.AuthMethod {
 //ImportSSHconfig imports the SSH configuration stored at the provided path.
 //Returns an error if the SSH configuration could not be instantiated.
 func ImportSSHconfig(config []byte) (SSHConfigs, error) {
+	//log.Infof("importing SSH config")
```
Remove this line.

ianvernon commented on test/helpers/ssh_command.go:
```diff
@@ -113,6 +124,7 @@ func ImportSSHconfig(config []byte) (SSHConfigs, error) {
 		hostConfig.user, _ = cfg.Get(key, "User")
 		hostConfig.port, _ = strconv.Atoi(port)
 		result[key] = &hostConfig
+		//log.Infof("generated SSHConfig for host %s: port: %d, user: %s, identity file: %s", hostConfig.host, hostConfig.port, hostConfig.user, hostConfig.identityFile)
```
Remove this line.

ianvernon commented on test/helpers/vagrant.go:
```diff
@@ -31,11 +32,12 @@ type Vagrant struct{}
-			//Sometimes Jenkins is not deleting the servers. So we need to make
-			//sure that we destroy before starts
+			// Always destroy if we are running in Jenkins. If not, just
+			// provisiong
```
Typo - just provision the VM.

ianvernon commented on test/k8sT/Policies.go:
```diff
@@ -50,24 +50,24 @@ var _ = Describe("K8sPolicyTest", func() {
 	BeforeEach(func() {
 		initialize()
-		// kubectl.CiliumPolicyDeleteAll("kube-system")
+		// kubectl.CiliumPolicyDeleteAll(helpers.KubeSystemNamespace)
```
Delete this line.

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -392,82 +412,84 @@ var _ = Describe("RunPolicies", func() {
 
 		//APP1 can connect to all Httpd1
-		connectivityTest(allRequests, "app1", "httpd1", BeTrue)
+		connectivityTest([]string{ping, ping6, http, http6}, helpers.App1, helpers.Httpd1, BeTrue)
```
Make this function use `allRequests`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -392,82 +412,84 @@ var _ = Describe("RunPolicies", func() {
 
 		// APP1 can reach using TCP HTTP2
-		connectivityTest(httpRequestsPublic, "app1", "httpd2", BeTrue)
+		connectivityTest([]string{http, http6}, helpers.App1, helpers.Httpd2, BeTrue)
```
Have this function use `httpRequestsPublic`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -392,82 +412,84 @@ var _ = Describe("RunPolicies", func() {
-		// APP3 can reach using TCP HTTP3, but can't ping EGRESS
-		connectivityTest(httpRequestsPublic, "app3", "httpd3", BeTrue)
+		// APP3 can reach using TCP HTTP2, but can't ping EGRESS
+		connectivityTest([]string{http, http6}, helpers.App3, helpers.Httpd3, BeTrue)
```
have this function use `httpRequestsPublic`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -392,82 +412,84 @@ var _ = Describe("RunPolicies", func() {
 
-		connectivityTest(allRequests, "app1", "httpd1", BeTrue)
-		connectivityTest(allRequests, "app2", "httpd1", BeTrue)
+		connectivityTest([]string{ping, ping6, http, http6}, helpers.App1, helpers.Httpd1, BeTrue)
```
have this function use `allRequests`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -392,82 +412,84 @@ var _ = Describe("RunPolicies", func() {
-		connectivityTest(allRequests, "app1", "httpd1", BeTrue)
-		connectivityTest(allRequests, "app2", "httpd1", BeTrue)
+		connectivityTest([]string{ping, ping6, http, http6}, helpers.App1, helpers.Httpd1, BeTrue)
+		connectivityTest([]string{ping, ping6, http, http6}, helpers.App2, helpers.Httpd1, BeTrue)
```
have this function use `allRequests`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -482,22 +504,22 @@ var _ = Describe("RunPolicies", func() {
 
-		connectivityTest(httpRequests, "app1", "httpd1", BeTrue)
-		connectivityTest(httpRequests, "app2", "httpd1", BeFalse)
+		connectivityTest([]string{http, http6}, helpers.App1, helpers.Httpd1, BeTrue)
```
have this function use `httpRequests`

ianvernon commented on test/runtime/Policies.go:
```diff
@@ -482,22 +504,22 @@ var _ = Describe("RunPolicies", func() {
-		connectivityTest(httpRequests, "app1", "httpd1", BeTrue)
-		connectivityTest(httpRequests, "app2", "httpd1", BeFalse)
+		connectivityTest([]string{http, http6}, helpers.App1, helpers.Httpd1, BeTrue)
+		connectivityTest([]string{http, http6}, helpers.App2, helpers.Httpd1, BeFalse)
```
have this function use `httpRequests`

ianvernon commented on test/runtime/connectivity.go:
```diff
@@ -48,230 +46,282 @@ var _ = Describe("RuntimeConnectivityTest", func() {
 		Expect(err).Should(BeNil())
 	}, 150)
 
+	dumpOutput := func(res *helpers.CmdRes) {
```
Remove this function and its uses, it was used for debugging. Functionality if added should be added into the docker.* functions themselves

ianvernon commented on test/runtime/lb.go:
```diff
@@ -39,26 +38,26 @@ var _ = Describe("RuntimeLB", func() {
 		initialized = true
 	}
 
 	containers := func(mode string) {
```
Add a TODO to rename this function; name is not clear.

ianvernon commented on test/runtime/monitor.go:
```diff
@@ -56,26 +68,26 @@ var _ = Describe("RuntimeMonitorTest", func() {
 	It("Cilium monitor verbose mode", func() {
 
-		res := cilium.Exec("config Debug=true DropNotification=true TraceNotification=true")
+		res := cilium.Exec("config MonitorDebug=true MonitorDropNotification=true MonitorTraceNotification=true")
```
This is incorrect; fix with `Sprintf`  and calls to the monitor configuration constants.

ianvernon reviewed (commented):
Left myself a self-review :)

nebril commented on test/helpers/cilium.go:
```diff
@@ -17,42 +17,44 @@ package helpers
 	"github.com/cilium/cilium/api/v1/models"
 
 	log "github.com/sirupsen/logrus"
+	"os"
```
Would be nice to have all stdlib imports in the first block.

ianvernon replied:
Good catch! :)

ianvernon replied:
Done.

nebril commented on test/helpers/cilium.go:
```diff
@@ -230,47 +239,61 @@ func (c *Cilium) GetFullPath(name string) string {
 }
 
-//PolicyGet return the policy value
+//PolicyGet runs `cilium policy get <id>`, where id is the name of a specific
```
Add space before PolicyGet

ianvernon replied:
Done.

nebril commented on test/helpers/kubectl.go:
```diff
@@ -288,32 +291,32 @@ func (kub *Kubectl) CiliumEndpointPolicyVersion(pod string) map[string]int64 {
 
 //CiliumExec runs cmd in the specified Cilium pod
```
Please add space before CiliumExec

ianvernon replied:
Done.

nebril commented on test/helpers/kubectl.go:
```diff
@@ -288,32 +291,32 @@ func (kub *Kubectl) CiliumEndpointPolicyVersion(pod string) map[string]int64 {
 	return kub.Node.Exec(cmd)
 }
 
 //CiliumNodesWait When a new cilium DS is set, the status can be ok but tunnels
```
Please add spaces after slashes.

ianvernon replied:
There needs to be a considerable amount of cleanup in the documentation of this code. To avoid having this PR balloon, I've created another issue to resolve these formatting and documentation issues: #2084. I went through and fixed the spacing and grammatical mistakes in `helpers/kubectl.go` and `helpers/cilium.go` which I will upload shortly. Is this OK?

ianvernon replied:
Done.

nebril commented on test/helpers/vagrant.go:
```diff
@@ -22,6 +22,7 @@ import (
 	"strings"
 
 	log "github.com/sirupsen/logrus"
+	//"regexp"
```
Please remove.

ianvernon replied:
Done.

nebril commented on test/helpers/wrappers.go:
```diff
@@ -0,0 +1,49 @@
+type PerfTest string
+
+const (
+	TCP_RR = PerfTest("TCP_RR")
```
What does RR stand for? Would be nice to have descriptive name here.

ianvernon replied:
Done. Added links to netperf documentation as well.

nebril commented on test/runtime/connectivity.go:
```diff
@@ -25,18 +22,19 @@ var _ = Describe("RuntimeConnectivityTest", func() {
 			return
 		}
 		logger = log.WithFields(log.Fields{"test": "RuntimeConnectivityTest"})
+		//logger.Level = log.InfoLevel
```
Please remove.

ianvernon replied:
Fixed.

nebril reviewed (changes requested):
Looks like great improvement overall, several nitpicks inline.

eloycoto commented on test/helpers/cilium.go:
```diff
@@ -173,17 +181,17 @@ func (c *Cilium) EndpointWaitUntilReady(validation ...bool) bool {
 			return true
 		}
-		logger.Infof("Endpoints are not ready valid='%d' invalid='%d'", valid, invalid)
+		logger.Infof("endpoints not ready; valid=%d invalid=%d", valid, invalid)
```
Could you use `log.WithFields`

ianvernon replied:
Done.

eloycoto commented on test/helpers/runtime.go:
```diff
@@ -15,11 +15,16 @@
 //runtime tests on the provided VM target and using logger log .
 func CreateNewRuntimeHelper(target string, log *log.Entry) (*Docker, *Cilium) {
-	return CreateDocker(target, log), CreateCilium(target, log)
+	fmt.Println("creating docker")
```
Why do you use `fmt.Println` instead of a log entry? It'll be not logged in the unittest output, it'll be a bit useless.

ianvernon replied:
Good catch, thanks! Done.

eloycoto commented on test/helpers/wrappers.go:
```diff
@@ -0,0 +1,49 @@
+
+// Ping6 returns the string representing the ping6 command to ping6 the
+// specified endpoint.
+func Ping6(endpoint string) string {
```
I never thought about this approach, awesome, thank you!

eloycoto commented on test/k8sT/Policies.go:
```diff
@@ -96,125 +95,130 @@ var _ = Describe("K8sPolicyTest", func() {
 
 		_, err = kubectl.Exec(
-			"default", appPods["app2"], fmt.Sprintf("curl http://%s/public", clusterIP))
+			helpers.DefaultNamespace, appPods[helpers.App2], fmt.Sprintf("curl http://%s/public", clusterIP))
```
Why do you not use `CurlFail` here?

ianvernon replied:
I only searched for curl commands which exactly matched `curl -s --fail --connect-timeout %d %s`. I will see if I can wrap all curl commands with `CurlFail`.

ianvernon replied:
Would you be OK with me adding this in a subsequent PR? I can file a ticket for this. I'd like to get this merged as soon as possible.

eloycoto commented on test/runtime/lb.go:
```diff
@@ -260,13 +275,25 @@ tc qdisc del dev lbtest2 clsact 2> /dev/null || true
-	err := helpers.RenderTemplateToFile("create_veth_interface", script, 0777)
+	scriptName := "create_veth_interface"
+	log.Infof("generating veth script: %s", scriptName)
+	err := helpers.RenderTemplateToFile(scriptName, script, 0777)
```
use here `os.ModePerm` instead 0777

eloycoto commented on test/runtime/monitor.go:
```diff
@@ -87,117 +100,117 @@ var _ = Describe("RuntimeMonitorTest", func() {
 		}
 
-		res := cilium.Exec("config Debug=true DropNotification=true TraceNotification=true")
+		res := cilium.Exec(fmt.Sprintf("config %s=true %s=true %s=true", MonitorDebug, MonitorDropNotification, MonitorTraceNotification))
```
Could you separate this line into 2 lines?

ianvernon replied:
Done.

eloycoto reviewed (changes requested):
Awesome work, a few minor comments.  ;-)

Commit:
helpers: fix documentation errors

Fixed multiple grammatical errors and clarified documentation for helper functions
in kubectl.go. Also ensured that spacing after `//`  was added for said comments.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
helpers: fix documentation errors

Fixed multiple grammatical errors and clarified documentation for helper functions
in cilium.go. Also ensured that spacing after `//`  was added for said comments.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
helpers: clean up documentation

Made formatting consistent in helpers/ssh_command.go and clarified some comments.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
helpers: misc. Cilium helper cleanup

* check whether ID is empty string in EndpointGet
* rename some variables for clarity
* add some more logs

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
helpers: add TODO for Ping wrapper

Signed-off by: Ian Vernon <ian@cilium.io>

joestringer commented on test/helpers/cilium.go:
```diff
@@ -81,49 +88,57 @@ func (c *Cilium) EndpointGet(id string) *models.Endpoint {
+	}, "endpoint not regenerated", &TimeoutConfig{Timeout: 100})
 	if err != nil {
-		logger.Errorf("Endpoint set failed:%s", err)
+		logger.Errorf("endpoint configuration update failed:%s", err)
```
`logger.WithError(err).Errorf(...)` ?

joestringer commented on test/helpers/cilium.go:
```diff
@@ -30,29 +30,31 @@ const (
 	return &Cilium{
 		Node:   node,
-		logCxt: log,
+		logger: log,
```
I couldn't see, but is anything responsible for adding `.WithField("VMname": vmName)` into this logger? Should it be done here?

joestringer commented on test/helpers/cilium.go:
```diff
@@ -173,39 +190,46 @@ func (c *Cilium) EndpointWaitUntilReady(validation ...bool) bool {
 	data := c.GetEndpoints()
 	if data.WasSuccessful() == false {
-		return nil, fmt.Errorf("Could't get endpoints")
+		return nil, fmt.Errorf("`cilium endpoint get` was not successful")
```
Tracing this back up to where the value came from, `c.GetEndpoints()` is actually calling `cilium endpoint list`. If we want the actual command being run here, then perhaps it should be propagated up from `GetEndpoints`. We might also / alternatively consider renaming `GetEndpoints` to `ListEndpoints`.

joestringer commented on test/helpers/cilium.go:
```diff
@@ -215,145 +239,169 @@ func (c *Cilium) GetEndpointsNames() ([]string, error) {
 	endpoints, err := c.GetEndpoints().Filter("{ [*].policy-enabled }")
 	if err != nil {
-		return result, fmt.Errorf("Can't get the endpoints")
+		return result, fmt.Errorf("cannot get endpoints")
```
What kind of err does `c.GetEndpoints().Filter()` return? Seems like we're dropping that err, but should we be?

aanm commented on test/helpers/cons.go:
```diff
@@ -20,9 +20,102 @@ import (
+
+	// Configuration options for endpoints. Copied from endpoint/endpoint.go
+	// TODO: these should be converted into types for use in configuration
+	// functions instead of using basic strings.
```
Can't you import directly from endpoint/endpoint.go?

ianvernon replied:
No, importing causes all of the transitive dependencies to be imported, which ends up trying to import CGo and thus requires kernel header dependencies. Because ginkgo runs directly on the host, importing will cause compilation failures for ginkgo on the mac. I can file a ticket to maybe factor out endpoint-related constants to a package, `pkg/endpoint/cons`, and then I'd be able to import that package. What do you think?

aanm replied:
```
 I can file a ticket to maybe factor out endpoint-related constants to a package, pkg/endpoint/cons, and then I'd be able to import that package. What do you think?
```
Yeah, it seems to be the only option.

aanm commented on test/helpers/cilium.go:
```diff
@@ -136,11 +142,12 @@ func (c *Cilium) EndpointWaitUntilReady(validation ...bool) bool {
 			return false
 		}
+		logger.Infof("cannot get endpoints: %s", err)
+		logger.Infof("sleeping 5 seconds and trying again to get endpoints")
```
`s/Infof/Info`

ianvernon replied:
Fixed.

aanm commented on test/helpers/cilium.go:
```diff
@@ -152,7 +159,7 @@ func (c *Cilium) EndpointWaitUntilReady(validation ...bool) bool {
 
 		if err := c.GetEndpoints().UnMarshal(&data); err != nil {
-			logger.Info("Can't get endpoints: %s", err)
+			logger.Info("cannot get endpoints: %s", err)
```
`s/Info/Infof/`

ianvernon replied:
Fixed.

aanm commented on test/helpers/cilium.go:
```diff
@@ -163,8 +170,9 @@ func (c *Cilium) EndpointWaitUntilReady(validation ...bool) bool {
-				if originalVal, _ := epsStatus[eps.ID]; len(eps.Status) <= originalVal {
-					logger.Infof("Endpoint '%d' is not regenerated", eps.ID)
+				// If the endpoint's latest statest message does not contain "Policy regeneration skipped", then it must be regenerating; wait until length of status message array changes.
+				if originalVal, _ := epsStatus[eps.ID]; !(len(eps.Status) > 0 && eps.Status[0].Message == "Policy regeneration skipped") && len(eps.Status) <= originalVal {
```
Write the assignment in 1 line and the `if` in other line

ianvernon replied:
Fixed.

aanm commented on test/helpers/node.go:
```diff
@@ -18,54 +18,59 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	log "github.com/sirupsen/logrus"
```
Import order

ianvernon replied:
Fixed.

aanm commented on test/helpers/wrappers.go:
```diff
@@ -0,0 +1,54 @@
+// Ping6 returns the string representing the ping6 command to ping6 the
+// specified endpoint.
+func Ping6(endpoint string) string {
+	return fmt.Sprintf("ping6  -c %d %s", PingCount, endpoint)
```
`s/ping6  -c/ping6 -c`

ianvernon replied:
Fixed.

aanm commented on test/helpers/cilium.go:
```diff
@@ -71,9 +71,13 @@ func (c *Cilium) Exec(cmd string) *CmdRes {
-	err := c.Exec(fmt.Sprintf("endpoint get %s", id)).UnMarshal(&data)
+	endpointGetCmd := fmt.Sprintf("endpoint get %s", id)
+	res := c.Exec(endpointGetCmd)
+	err := res.UnMarshal(&data)
```
`s/UnMarshal/Unmarshal/`

ianvernon replied:
Fixed.

aanm commented on test/helpers/wrappers.go:
```diff
@@ -32,6 +32,7 @@ const (
 // Ping returns the string representing the ping command to ping the specified
 // endpoint.
 func Ping(endpoint string) string {
+	// TODO: add W -2 ?
```
Maybe `W -2` for Pingfail and `W -5` for this case?

ianvernon replied:
I can do this as a follow-up PR? The current functionality works OK, this was more of a future thing to look into.

Commit:
test: rename UnMarshal to Unmarshal

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: change log message Infof --> Info

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: split assignment and if condition

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
test/helpers: fix import order in node.go

Signed-off by: Ian Vernon <ian@cilium.io>

joestringer commented on test/helpers/cilium.go:
```diff
@@ -215,145 +239,169 @@ func (c *Cilium) GetEndpointsNames() ([]string, error) {
 	// We check before setting PolicyEnforcement; if we do not, EndpointWait
 	// will fail due to the status of the endpoints not changing.
-	res := c.Exec("config | grep PolicyEnforcement | awk '{print $2}'")
+	log.Infof("setting PolicyEnforcement=%s", status)
```
If the line immediately below is replacing the hardcoded PolicyEnforcement string with a variable, perhaps this one should too?

joestringer commented on test/helpers/cmd.go:
```diff
@@ -26,52 +26,67 @@ import (
+	params []string      // Parameters to provide to command
+	stdout *bytes.Buffer // Stdout from running cmd
+	stderr *bytes.Buffer // Stderr from running cmd
+	exit   bool          // Whether command successfully ran.
```
It's still not immediately clear which case is success and which is fail. This could be further improved by either renaming it `success`, or documenting that true is success (assuming that's actually the case).

joestringer commented on test/helpers/kubectl.go:
```diff
@@ -286,42 +304,45 @@ func (kub *Kubectl) CiliumEndpointPolicyVersion(pod string) map[string]int64 {
-//are not in place yet. Checking the `kubectl get nodes` we can know if cilium
-//added the node correctly. In case of timeout will return an error
+// CiliumNodesWait waits up until the specified timeout has elapsed
+// until all nodes in the Kubernetes cluster are annotated  with Cilium
```
I found the above run-on combination of descriptions a little confusing. The function doesn't take a timeout, so maybe describing it as 'specified' is a little unclear.  Here's an attempt, feel free to take it or parts of it or just leave it alone if you don't think it helps:

```
CiliumNodesWait waits until all nodes in the Kubernetes cluster are annotated with Cilium 
annotations.  Its runtime is bounded by a maximum of `HelperTimeout`. When a node is 
annotated with Cilium annotations, it indicates...
```

joestringer commented on test/helpers/runtime.go:
```diff
@@ -21,5 +21,9 @@ import (
 func CreateNewRuntimeHelper(target string, log *log.Entry) (*Docker, *Cilium) {
-	return CreateDocker(target, log), CreateCilium(target, log)
+	log.Infof("creating docker")
+	docker := CreateDocker(target, log)
```
I'm a little surprised that these log messages are being added in `CreateNewRuntimeHelper()`, and not `CreateDocker()`.

ianvernon replied:
I'll remove them. I have a plan to remove the concept of "Docker" and "Cilium" helpers in a follow-up PR, as the different types don't provide much value and have identical members.

joestringer commented on test/helpers/wrappers.go:
```diff
@@ -0,0 +1,55 @@
+// Ping returns the string representing the ping command to ping the specified
+// endpoint.
+func Ping(endpoint string) string {
+	// TODO: add W -2 ?
```
:+1:

joestringer commented on test/runtime/lb.go:
```diff
@@ -260,13 +275,25 @@ tc qdisc del dev lbtest2 clsact 2> /dev/null || true
+	scriptPath := fmt.Sprintf("%s%s", helpers.BasePath, scriptName)
+
+	ipAddrCmd := "sudo ip addr add fd02:1:1:1:1:1:1:1 dev cilium_host"
+	res := node.Exec(ipAddrCmd)
```
Does this also check the output? At a glance it looks like the `ExpectSuccess()` was dropped, which may allow the execution to continue despite a failure.

ianvernon replied:
If you run the test multiple times locally, the command will fail. I can't see why it would ever not fail in the context of the tests, so I decided to remove it.

joestringer reviewed (commented):
LGTM overall, I left a few comments that could be addressed either here or in a followup.
